app/src/main/java/uk/gov/onelogin/signOut/domain/SignOutUseCase.kt

```diff
@@ -1,26 +1,30 @@
 package uk.gov.onelogin.signOut.domain
 
+import androidx.fragment.app.FragmentActivity
 import javax.inject.Inject
 import uk.gov.onelogin.login.biooptin.BiometricPreferenceHandler
 import uk.gov.onelogin.tokens.usecases.RemoveAllSecureStoreData
 import uk.gov.onelogin.tokens.usecases.RemoveTokenExpiry
+import uk.gov.onelogin.wallet.DeleteWalletDataUseCase
 
 fun interface SignOutUseCase {
-    operator fun invoke()
+    suspend operator fun invoke(activityFragment: FragmentActivity)
```

Comment:
Please refer to my large rant in DeleteWalletDataUsecase, as the same applies here. You could argue that it was already in use and therefore keeping it that way is ok, but I believe in refactoring as you go, and as changes are being made then the new code should better than the old code.